app/main/views/suppliers.py

```diff
@@ -82,65 +86,73 @@ def dashboard():
 
 @main.route('/edit', methods=['GET'])
 @login_required
-def edit_supplier(supplier_form=None, contact_form=None, error=None):
+def edit_supplier(supplier=None, errors=None):
+    if not supplier:
+        try:
+            supplier = data_api_client.get_supplier(
+                current_user.supplier_code
+            )['supplier']
+        except APIError as e:
+            current_app.logger.error(e)
+            abort(e.status_code)
+
+    businessDetails = supplier
+    for link in businessDetails.get('extraLinks', []):
+        if link['label'] == 'LinkedIn':
+            businessDetails['linkedin'] = link['url']
+    form = DmForm()
+    rendered_component = render_component('bundles/SellerRegistration/BusinessDetailsWidget.js',
+                                          {'form_options': {'csrf_token': form.csrf_token.current_token,
+                                                            'errors': errors},
+                                           'form': {'businessDetails': businessDetails}})
+    return render_template(
+        '_react.html',
+        component=rendered_component
+    )
+
+
+@main.route('/edit', methods=['POST'])
+@login_required
+def update_supplier():
     try:
         supplier = data_api_client.get_supplier(
```

Comment:
i actually don't think we need to do this

apiclient.update_supplier() does a patch - it will just replace the fields that you send it